include/param.h

```diff
@@ -294,14 +294,17 @@ class Params
    * @param id The ID of the parameter
    * @return The value of the parameter
    */
-  inline const int get_param_int(uint16_t id) const { return params.values[id]; }
+  inline int get_param_int(uint16_t id) const { return params.values[id]; }
 
   /**
    * @brief Get the value of a floating point parameter by id
    * @param id The ID of the parameter
    * @return The value of the parameter
    */
-  inline const float get_param_float(uint16_t id) const { return *(float *) &params.values[id]; }
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
+  inline float get_param_float(uint16_t id) const { return reinterpret_cast<const float &>(params.values[id]); }
```

Comment:
I'm going to see if I can fix this by switching the param stuff to using a `union` instead of casting... I'd like to get rid of as many of these `GCC diagnostic ignored` sections as possible. Ideally, I'd like to get rid of all of them in our flight stack and only use it for mavlink and breezy.